sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -166,6 +176,23 @@ export interface KeyVaultSelectiveKeyRestoreResult {
     startTime: Date;
 }
 
+// @public
+export type KeyVaultSetting = UnknownKeyVaultSetting | BooleanKeyVaultSetting;
+
+// @public
+export interface KeyVaultSettingCommon {
+    name: string;
+}
+
+// @public
+export class KeyVaultSettingsClient {
+    constructor(vaultUrl: string, credential: TokenCredential, options?: SettingsClientOptions);
+    getSetting(settingName: string, options?: GetSettingOptions): Promise<KeyVaultSetting>;
+    listSettings(options?: ListSettingsOptions): Promise<ListSettingsResponse>;
+    updateSetting(settingName: string, value: boolean, options?: UpdateSettingOptions): Promise<KeyVaultSetting>;
```

Comment:
We made the `KeyVaultSetting` mutable and pass that to `updateSetting` now. See internal Teams discussion for more information. Is there reason to be different i.e., to be idiomatic?

Reply:
We can iron this out more before GA. I'm not sure the concept of making `KeyVaultSetting` mutable maps very well to JS (since our `KeyVaultSetting` is just a plain object, the user can just mutate it as-is!), but we can discuss whether updating the signature of `updateSetting` to take in something that looks like a `KeyVaultSetting` makes sense.

Reply:
I like the idea of passing `KeyVaultSetting` as is here, so you can mutate and roundtrip the object. Otherwise we have to add overloads for the different values later, which could get messy since we have no way to know if the value they are passing is correct for a given name

Reply:
For language that have unions like TS, overloads are something to consider though, if that's more idiomatic. For example,

```typescript
updateSetting(name: string, value: string | boolean, options: UpdateSettingsOptions?): Promise<KeyVaultSettings>;
```

Your call. With .NET - and Java, for now, even though they support unions - we just didn't want to add a bunch of overloads. Consolidating type conversions to a single model was only part of the reasoning.